cachito/workers/tasks/general.py

```diff
@@ -169,6 +171,51 @@ def save_bundle_archive_checksum(request_id: int) -> None:
     bundle_dir.bundle_archive_checksum.write_text(checksum, encoding="utf-8")
 
 
+def _check_packages_data_on_api(
+    request_id: int, packages_count: int, dependencies_count: int
+) -> bool:
+    request = get_request(request_id)
+    actual_packages_count = len(request.get("packages", []))
+    actual_dependencies_count = len(request.get("dependencies", []))
+
+    if actual_packages_count == packages_count and actual_dependencies_count == dependencies_count:
+        return True
+
+    if actual_packages_count != 0 or actual_dependencies_count != 0:
+        log.warning(
+            "Mismatch in the number of expected packages while checking the written data."
+            f"Expected {packages_count} packages, got {actual_packages_count}."
+            f"Expected {dependencies_count} dependencies, got {actual_dependencies_count}."
+        )
+
+    return False
+
+
+def _wait_until_packages_file_is_available(
+    request_id: int, packages_count: int, dependencies_count: int
+) -> None:
+    config = get_worker_config()
+    interval = config.cachito_finalize_request_packages_check_interval
+    max_attempts = config.cachito_finalize_request_packages_check_max_attempts
+
+    if max_attempts == 0:
+        return
+
+    attempts = 0
+    check_passed = False
+
+    while not check_passed and attempts < max_attempts:
+        if attempts > 0:
+            time.sleep(interval)
+
+        log.info("Checking if the packages file was properly loaded (attempt %i).", attempts)
```

Comment:
Should the first attempt be logged as "attempt 1" rather than 0?

Reply:
Sorry, this is Python, not Lua :(

Jokes apart, most definitely, will fix it right now.